# O conjunto de dados vgsales.csv contém informações sobre vendas de videogames

# Etapa de coleta de dados:
## A coleta de dados é a parte cujo objetivo é reunir as informações relevantes para análise.<br><br>fonte para baixar o arquivo: https://www.kaggle.com/datasets/dandanjia/vgsales-csv

# Etapa de preparação dos dados

In [3]:
import pandas as pd

### Carregando o conjunto de dados vgsales.csv usando Pandas.

In [6]:
df = pd.read_csv('vgsales.csv', sep=',', encoding='latin-1')

In [7]:
df.head(n=5)

Rank                      Name Platform    Year         Genre Publisher  \
0     1                Wii Sports      Wii  2006.0        Sports  Nintendo   
1     2         Super Mario Bros.      NES  1985.0      Platform  Nintendo   
2     3            Mario Kart Wii      Wii  2008.0        Racing  Nintendo   
3     4         Wii Sports Resort      Wii  2009.0        Sports  Nintendo   
4     5  Pokemon Red/Pokemon Blue       GB  1996.0  Role-Playing  Nintendo   

   NA_Sales  EU_Sales  JP_Sales  Other_Sales  Global_Sales  
0     41.49     29.02      3.77         8.46         82.74  
1     29.08      3.58      6.81         0.77         40.24  
2     15.85     12.88      3.79         3.31         35.82  
3     15.75     11.01      3.28         2.96         33.00  
4     11.27      8.89     10.22         1.00         31.37

- Rank: Posição de venda
- Name: Nome do jogo
- Platform: Plataforma do jogo
- Year: Ano de lançamento
- Genre: Gênero do jogo
- Publisher: Editora do jogo
- NA_Sales: Vendas na América do Norte (em milhões)
- EU_Sales: Vendas na Europa (em milhões)
- JP_Sales: Vendas no Japão (em milhões)
- Other_Sales: Vendas em outras regiões (em milhões)
- Global_Sales: Vendas globais (em milhões)

In [10]:
df

Rank                                              Name Platform  \
0          1                                        Wii Sports      Wii   
1          2                                 Super Mario Bros.      NES   
2          3                                    Mario Kart Wii      Wii   
3          4                                 Wii Sports Resort      Wii   
4          5                          Pokemon Red/Pokemon Blue       GB   
...      ...                                               ...      ...   
16593  16596                Woody Woodpecker in Crazy Castle 5      GBA   
16594  16597                     Men in Black II: Alien Escape       GC   
16595  16598  SCORE International Baja 1000: The Official Game      PS2   
16596  16599                                        Know How 2       DS   
16597  16600                                  Spirits & Spells      GBA   

         Year         Genre   Publisher  NA_Sales  EU_Sales  JP_Sales  \
0      2006.0        Sports    Nintendo     41.49     29.02      3.77   
1      1985.0      Platform    Nintendo     29.08      3.58      6.81   
2      2008.0        Racing    Nintendo     15.85     12.88      3.79   
3      2009.0        Sports    Nintendo     15.75     11.01      3.28   
4      1996.0  Role-Playing    Nintendo     11.27      8.89     10.22   
...       ...           ...         ...       ...       ...       ...   
16593  2002.0      Platform       Kemco      0.01      0.00      0.00   
16594  2003.0       Shooter  Infogrames      0.01      0.00      0.00   
16595  2008.0        Racing  Activision      0.00      0.00      0.00   
16596  2010.0        Puzzle    7G//AMES      0.00      0.01      0.00   
16597  2003.0      Platform     Wanadoo      0.01      0.00      0.00   

       Other_Sales  Global_Sales  
0             8.46         82.74  
1             0.77         40.24  
2             3.31         35.82  
3             2.96         33.00  
4             1.00         31.37  
...            ...           ...  
16593         0.00          0.01  
16594         0.00          0.01  
16595         0.00          0.01  
16596         0.00          0.01  
16597         0.00          0.01  

[16598 rows x 11 columns]

## Estatísticas Descritivas: Calcule as estatísticas descritivas para as colunas de vendas<br><br> (NA_Sales, EU_Sales, JP_Sales, Other_Sales, Global_Sales).
### As estatísticas descritivas incluem contagens, média, desvio padrão, valores mínimos e máximos para cada coluna numérica

In [11]:
df.describe() # Usando para colunas numéricas

Rank          Year      NA_Sales      EU_Sales      JP_Sales  \
count  16598.000000  16327.000000  16598.000000  16598.000000  16598.000000   
mean    8300.605254   2006.406443      0.264667      0.146652      0.077782   
std     4791.853933      5.828981      0.816683      0.505351      0.309291   
min        1.000000   1980.000000      0.000000      0.000000      0.000000   
25%     4151.250000   2003.000000      0.000000      0.000000      0.000000   
50%     8300.500000   2007.000000      0.080000      0.020000      0.000000   
75%    12449.750000   2010.000000      0.240000      0.110000      0.040000   
max    16600.000000   2020.000000     41.490000     29.020000     10.220000   

        Other_Sales  Global_Sales  
count  16598.000000  16598.000000  
mean       0.048063      0.537441  
std        0.188588      1.555028  
min        0.000000      0.010000  
25%        0.000000      0.060000  
50%        0.010000      0.170000  
75%        0.040000      0.470000  
max       10.570000     82.740000

- 1ª linha é o número de casos (sendo o total de 16598) Obs.: a coluna "Year" não está completa porque existe dados nulos
- 2ª linha a média de cada coluna
- 3ª linha desvio padrão
- 4ª linha valor mínimo
- 5ª, 6ª, 7ª linhas são os quartis: dividem o conjunto de dados em 4 partes, ou seja, em pedaços de tamanhos iguais que contém 25%, 50% e 75% dos dados
- 8ª linha valor máximo

In [16]:
df.isnull().sum() # Quantidade de dados nulos

Rank              0
Name              0
Platform          0
Year            271
Genre             0
Publisher        58
NA_Sales          0
EU_Sales          0
JP_Sales          0
Other_Sales       0
Global_Sales      0
dtype: int64

#### A coluna "Year" tem 271 linhas com dados nulos

## Total de Vendas por Gênero: Calcule o total de vendas globais para cada gênero.

In [19]:
total_vendas_por_genero = df.groupby(['Genre'])['Global_Sales'].sum()

In [20]:
total_vendas_por_genero

Genre
Action          1751.18
Adventure        239.04
Fighting         448.91
Misc             809.96
Platform         831.37
Puzzle           244.95
Racing           732.04
Role-Playing     927.37
Shooter         1037.37
Simulation       392.20
Sports          1330.93
Strategy         175.12
Name: Global_Sales, dtype: float64

## Total de Vendas por Plataforma: Calcule o total de vendas globais para cada plataforma.

In [21]:
total_vendas_por_plataforma = df.groupby(['Platform'])['Global_Sales'].sum()

In [22]:
total_vendas_por_plataforma

Platform
2600      97.08
3DO        0.10
3DS      247.46
DC        15.97
DS       822.49
GB       255.45
GBA      318.50
GC       199.36
GEN       28.36
GG         0.04
N64      218.88
NES      251.07
NG         1.44
PC       258.82
PCFX       0.03
PS       730.66
PS2     1255.64
PS3      957.84
PS4      278.10
PSP      296.28
PSV       61.93
SAT       33.59
SCD        1.87
SNES     200.05
TG16       0.16
WS         1.42
Wii      926.71
WiiU      81.86
X360     979.96
XB       258.26
XOne     141.06
Name: Global_Sales, dtype: float64

## Jogo Mais Vendido: Determine qual jogo teve as maiores vendas globais.

In [30]:
total_vendas_por_nome_jogo = df.groupby(['Name'])['Global_Sales'].sum()
#jogo_mais_vendido_vandas_globais = total_vendas_por_nome_jogo.groupby('Name').idxmax().reset_index()

In [31]:
total_vendas_por_nome_jogo

Name
'98 Koshien                                0.41
.hack//G.U. Vol.1//Rebirth                 0.17
.hack//G.U. Vol.2//Reminisce               0.23
.hack//G.U. Vol.2//Reminisce (jp sales)    0.16
.hack//G.U. Vol.3//Redemption              0.17
                                           ... 
thinkSMART: Chess for Kids                 0.01
uDraw Studio                               2.46
uDraw Studio: Instant Artist               0.21
wwe Smackdown vs. Raw 2006                 3.00
¡Shin Chan Flipa en colores!              0.14
Name: Global_Sales, Length: 11493, dtype: float64

In [32]:
df['Name']

0                                              Wii Sports
1                                       Super Mario Bros.
2                                          Mario Kart Wii
3                                       Wii Sports Resort
4                                Pokemon Red/Pokemon Blue
                               ...                       
16593                  Woody Woodpecker in Crazy Castle 5
16594                       Men in Black II: Alien Escape
16595    SCORE International Baja 1000: The Official Game
16596                                          Know How 2
16597                                    Spirits & Spells
Name: Name, Length: 16598, dtype: object